src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -197,6 +198,15 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, HttpsConn
         listenOptions.IsTls = true;
         listenOptions.HttpsOptions = httpsOptions;
 
+        if (httpsOptions.TlsClientHelloBytesCallback is not null)
```

Comment:
Need to add this for the other UseHttps overload at the bottom of the file?

Reply:
I think it has a different use case (allows setup via `TlsHandshakeCallbackOptions`), but I added a check `if (listenOptions.HttpsOptions?.TlsClientHelloBytesCallback is not null)`

> Configure Kestrel to use HTTPS. This does not use default certificates or other defaults specified via config or
    KestrelServerOptions.ConfigureHttpsDefaults(Action<HttpsConnectionAdapterOptions>)

Reply:
But it does raise the question about client hello in that case. It also shows we have a client hello API already (although not very good) https://source.dot.net/#Microsoft.AspNetCore.Server.Kestrel.Core/TlsHandshakeCallbackContext.cs,97120719b6888c33,references

Reply:
I tried to move the callback to `TlsHandshakeCallbackOptions`, but it **requires** setting [OnConnection](https://source.dot.net/#Microsoft.AspNetCore.Server.Kestrel.Core/TlsHandshakeCallbackOptions.cs,19), and I dont think it should be **required** for TLS client hello.

I personally would leave it where we designed it - in `HttpsConnectionAdapterOptions`. Let me know what you think!

(talked with @halter73 about it as well)

Reply:
Actually, I removed registration from overload using `TlsHandshakeCallbackOptions`, because it can result in 2 registrations of the middleware